### DISCUSSION TITLE
hid_generic_inout: how to receive 3 Bytes from host ?

### DISCUSSION THREAD

**Discussion by rapapar** [hathach/tinyusb#1902] [Q&A]
Hello,
May someone here help to let me know how to simply get the 3 first BYTES of the received messages from host ?
In my example, I need ReceivedDataBuffer[]  with index 0,1,2 only from my computer ...

I show you an edited file from the file tinyusb\examples\device\hid_generic_inout\src\main.c
It can not compile it is just "ideas"
Im' not even sure that I can get data in tud_hid_get_report_cb, 
I can't find doc, so I read all the examples but it is too complicated for me.

```
// Invoked when received GET_REPORT control request
uint16_t tud_hid_get_report_cb(uint8_t itf, uint8_t report_id, hid_report_type_t report_type, uint8_t *buffer, uint16_t reqlen)
{

unsigned int  ReceivedDataBuffer[];
 // SOMETHING IS MISSING HERE to fullfill ReceivedDataBuffer, here you may help me please

  if (ReceivedDataBuffer[0] == 0x26)
  {
    unsigned int TempValue, LOW, HIGH;
    TempValue = (ReceivedDataBuffer[2] << 8) + ReceivedDataBuffer[1];
    // need it on 10bits
    LOW = (TempValue & 0xff) << 6; // 00000011 to  11000000
    HIGH = (((TempValue >> 8) & 0xff) << 6) | (TempValue & 0xff) >> 2;
  }

  return 0;
}

```

**Comment by inbed-io-dmberry** [comment]
uint8_t *buffer is the data buffer that is passed onto the call back, 

You can reference it directly, so 

if (ReceivedDataBuffer[0] == 0x26) 
would be
if (buffer[0] == 0x26)

**Comment by rapapar** [comment]
Ok but I've already have try that.
After many tests and debug, it appears that this function get_report is never called.
I've to work in the set_report function.

Thanks

**Reply by hathach** [reply]
get_report() is when host send GET_REPORT request i.e getting report from device. Yours indeed need set_report()
